Wikipedia/Code/MWKDataStore.m

```diff
@@ -467,10 +464,24 @@ - (void)migrateToQuadKeyLocationIfNecessaryWithCompletion:(nonnull void (^)(NSEr
 
 #pragma mark - Background Contexts
 
+- (void)backgroundContextDidSave:(NSNotification *)note {
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
It might be that this can't be async and should not allow this method to return until the information is read from the notification. Also, it'd be ideal if this didn't happen on the main thread and had access limited by a semaphore.